src/app/content/highlights/reducer.ts

```diff
@@ -194,7 +194,16 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
       };
     }
     case getType(actions.focusHighlight): {
-      return {...state, currentPage: { ...state.currentPage, hasUnsavedHighlight: false, focused: action.payload }};
+      const isRefocusing = state.currentPage.focused === action.payload;
+      return {
+        ...state,
+        currentPage: {
+          ...state.currentPage,
+          focused: action.payload,
+          hasUnsavedHighlight: isRefocusing,
```

Comment:
i think what you want to do is pull forward the previous value unless the id is different, in which cases set it to false. in this change you're assuming that a refocus happens when there are pending changes, which may be true now but seems like a weird assumption.